packages/resolve-redux/src/subscribe_adapter.js

```diff
@@ -4,8 +4,11 @@ import { getRootableUrl } from './util'
 
 export default function subscribeAdapter() {
   let onEvent, onDisconnect
-
-  const socket = socketIOClient(window.location.origin, {
+  const origin =
+    typeof navigator !== 'undefined' && navigator.product === 'ReactNative'
+      ? process.env['ROOT_DIR']
```

Comment:
Maybe segregate root directory source via classification for absolute/relative URL supposed in `ROOT_DIR` environment variable? Aka check for `ROOT_DIR` content, and if it's absolute URL path, then take origin part from here, and instead take `window.location.origin` value.
It maybe useful for use cases, when origin for subscription should differ relative to current origin. Also, `path` variable can be determined such way.